<?xml version="1.0"?>
<launch>
	<!--Generate relative subject_pose and task_waypose-->
	<node pkg="vicon_pose" type="vicon_pose" name="conv_q_p_Controller">
		<param name="object_name" value="wand" type="str" />
		<remap from= "subject_pose" to="controller_pose" />
	</node>
	
	<node pkg="vicon_pose" type="vicon_pose" name="conv_q_p_UAV">
		<param name="object_name" value="CRASH" type="str" />
		<remap from= "subject_pose" to="UAV_pose" />
	</node>

	<node pkg="demo_UAV_controller" type="demo_UAV_controller" name="demo_UAV_controller">
		<remap from= "subject_pose" to="relative_subject_pose" />
		<remap from= "task_waypose" to="relative_task_waypose" />
	</node>

	<!--Generate global subject_pose and task_waypose-->
	<node pkg="vicon_pose" type="vicon_pose" name="conv_q_p_global">
		<param name="object_name" value="CRASH" type="str" />
		<remap from= "subject_pose" to="global_subject_pose" />
	</node>
	<node name="script_input" type="script_input" pkg="script_input">
		<param name="task_pub_rate" value="20.0" />
 		<param name="pose_error" value="0.5" />
		<param name="dir_error" value="0.3" />
		<param name="land_delay" value="10.0" />
	</node>
	<node name="gui_fly" type="gui_fly" pkg="gui_fly">
		<remap from= "task_waypose" to="global_task_waypose" />
	</node>

	<!--Select the subject_pose and task_waypose based on the parameter pose_source (1 for global, 2 for relative)-->
	<node name="subject_pose_arb" type="arb_subject_ctrl" pkg="arbitration">
		<remap from="shape_shifter" to="subject_pose"/>
		<rosparam param="pose_source">
                [
                [ "global_subject_pose", [ 1 ] ],
                [ "relative_subject_pose", [ 2 ] ],
                ]
                </rosparam>
	</node>
	<node name="task_waypose_arb" type="arb_subject_ctrl" pkg="arbitration">
		<remap from="shape_shifter" to="task_waypose"/>
		<rosparam param="pose_source">
                [
                [ "global_task_waypose", [ 1 ] ],
                [ "relative_task_waypose", [ 2 ] ],
                ]
                </rosparam>
	</node>

	<!--Set communication method-->
	<node name="robot_comm" type="serial_comm" pkg="communication">
		<remap from="serial_rx_data" to="robot_rx_data"/>
		<remap from="serial_tx_data" to="robot_tx_data"/>
		<param name="serial_device" value="/dev/ttyUSB0"/>
    	</node>
  
	<!--Launch mitAscTec-->
	<include  file="$(find collab_launch)/new_pid_ctrl.launch" />
	<include  file="$(find collab_launch)/common.launch" />
</launch>
